packages/base/src/EventProvider.js

```diff
@@ -19,13 +19,13 @@ class EventProvider {
 
 	detachEvent(eventName, fnFunction) {
 		const eventRegistry = this._eventRegistry;
-		let eventListeners = eventRegistry.get(eventName);
+		const eventListeners = eventRegistry.get(eventName);
 
 		if (!eventListeners) {
 			return;
 		}
-
-		eventListeners = eventListeners.filter(fn => fn !== fnFunction);
+		const indexOfFnToDetach = eventListeners.indexOf(fnFunction);
+		if(~indexOfFnToDetach) eventListeners[eventName].splice(indexOfFnToDetach, 1);
```

Comment:
I checked that this won't pass the linter:
```
  29:3  error  Expected { after 'if' condition                            curly
  29:3  error  Expected space(s) after "if"                               keyword-spacing
  29:6  error  Unexpected use of '~'                                      no-bitwise
```

Please use `indexOfFnToDetach !== -1` - it is much easier to read and let minifiers figure out an equivalent with less characters.